clash-ghc/src-ghc/Clash/GHC/PartialEval/Eval.hs

```diff
@@ -603,7 +603,7 @@ apply val arg = do
       error ("apply: Cannot apply " <> show arg <> " to " <> show f)
  where
   -- Somewhat of a cheat, but very quick to implement.
-  valueType tcm = termType tcm . asTerm
```

Comment:
This is actually why I wanted this and the `HasFreeVars` class in the first place, but I intended to make those instances in a PE branch instead of here

Reply:
TODO: add TODO